src/cloud-api-adaptor/install/overlays/ibmcloud/kustomization.yaml

```diff
@@ -17,6 +17,9 @@ configMapGenerator:
   namespace: confidential-containers-system
   literals:
   - CLOUD_PROVIDER="ibmcloud"
+  - ENABLE_CLOUD_PROVIDER_EXTERNAL_PLUGIN="false" # flag to enable/disable dynamically loaded cloud provider external plugins feature
+  - CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH="/run/cloud-api-adaptor/plugins/libvirt.so" # caa will try to load the cloud provider external plugin from this path in runtime
```

Comment:
nit: CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH should be empty for all the overlays

Reply:
updated to for all the overlays :
```
  - ENABLE_CLOUD_PROVIDER_EXTERNAL_PLUGIN="false" # flag to enable/disable dynamically loaded cloud provider external plugin feature
  - # CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH="" # Uncomment and set if you want to dynamically load a cloud provider external plugin from this path
  - # CLOUD_PROVIDER_EXTERNAL_PLUGIN_HASH="" # Uncomment and set sha256sum of "${CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH}"

```

Reply:
@liudalibj do you want to include a separate overlay directory for external-plugin ? That can include the uncommented hostPath and other configs based on the example external plugin.
Just a thought.

Reply:
@bpradipt 
In our the code logical, the new added directory should be "cloud-provider" name. it's not good to add a new directory here.
Currently we didn't have steps to setup peerpod env with the example external plugin directly.
We can consider use a simple way to setup peerpod env with example external plugin directly.

Maybe we can:
1. Keep the comment codes for all the built-int overlays now.
2. Remove the comment codes directly

Both option is fine to me, WDYT?

Reply:
Sorry, I just want to check if I understand the question here. Is that that rather than editing the build-it provider's kustomization.yaml to have a commented out:
```
#- CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH="" # Uncomment and set if you want to dynamically load a cloud provider external plugin from this path
  #- CLOUD_PROVIDER_EXTERNAL_PLUGIN_HASH="" # Uncomment and set sha256sum of "${CLOUD_PROVIDER_EXTERNAL_PLUGIN_PATH}"
```
We instead rely on a separate `install/overlays/external.kustomization.yaml` with it in?

I think this is desirable. Would this help improve the user experience too? At the moment we have to create a "normal" peer pods install and then modify the peer-pods-cm config, log into the worker and then modify the daemonset and deployments that is fairly tricky to do.

I appreciate that this is probably beyond the scope of this PR and might need some sort of dummy provider, or the ability to start with no cloud provider, which might cause more problems than it solves though?

Reply:
In the short term maybe deleting the comments from the kustomization.yamls might be the best option to not confuse standard users?

Reply:
Sounds good to me. Having the instructions documented is good enough for now.

Reply:
@stevenhorsman what I meant was to have an overlay directory `external` for the external provider, so that deployment is similar to any other provider and doesn't require manual editing. However @liudalibj has a point. The cloud provider name is same as the plugin provider name, so doesn't really make sense to have an `external` directory.

Reply:
deleted the comments from the kustomization.yamls